compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -588,6 +589,10 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             return true;
         }
 
+        if (sourceTag == TypeTags.NEVER && targetTag == TypeTags.XML_TEXT) {
```

Comment:
Shouldn't this be true only when it is part of xml? Probably part of `isXMLTypeAssignable`.

The following is also allowed due to this change.

```ballerina
    never[] arr = [];
    'xml:Text text = arr[0]; // should fail
```